tests/integration/java-integration/src/test/java/io/fluentbit/test/LogEntry.java

```diff
@@ -0,0 +1,36 @@
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2019-2021 The Fluent Bit Authors
+ *  Copyright (C) 2015-2018 Treasure Data Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.fluentbit.test;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * The tests log key-value pairs. Key is always data, value is variable. This class represents this structure.
+ */
+public class LogEntry {
+    @JsonProperty
+    private String data;
+
+    public String getData() {
+        return data;
+    }
+
+    public void setData(String data) {
```

Comment:
Is there any issue with different string encodings? Feels like it could be final too.

Reply:
Java Strings are always Unicode.  It gets converted to the OS default encoding, which should be in docker UTF8.

Test -> FluentBit: happens via the fluency library
FluentBit -> Logging Destination: LoggingDestinationServer.java: 
`BufferedReader in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()));`

Can be changed to something like
```
BufferedReader in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream(), Charset.forName("IBM1141")));```

Reply:
Java strings are, my query was if this was ever consuming non-Java "strings" (e.g. C-style which are just null-terminated ASCII single byte character arrays and not strings).

As long as it works on non-US/UK encodings I'm happy.

Reply:
No it doesn't. The tests define the data being logged. Currently it is just A-Z, a-z and 0-9.